source/common/tcp_proxy/upstream.h

```diff
@@ -171,11 +171,11 @@ class HttpUpstream : public GenericUpstream, protected Http::StreamCallbacks {
     // Http::ResponseDecoder
     void decode1xxHeaders(Http::ResponseHeaderMapPtr&&) override {}
     void decodeHeaders(Http::ResponseHeaderMapPtr&& headers, bool end_stream) override {
+         parent_.config_.propagateResponseHeaders(std::move(headers),
+         parent_.downstream_info_.filterState());
       if (!parent_.isValidResponse(*headers) || end_stream) {
```

Comment:
I suggest the change to be the following:
      bool isValidResponse = parent_.isValidResponse(*headers);
      parent_.config_.propagateResponseHeaders(std::move(headers),
                                               parent_.downstream_info_.filterState());
      if (!isValidResponse || end_stream) {...}
Because we use std::move(headers) in propagateResponseHeaders

Reply:
Thank you! Done